vm/src/pyobject.rs

```diff
@@ -509,11 +512,28 @@ impl PyObject<dyn PyObjectPayload> {
             let ret = unsafe { PyRc::from_raw(ptr) };
             Ok(ret)
         } else {
-            Err(self)
+            Err(self.into())
         }
     }
 }
 
+impl PyObjectRc {
```

Comment:
Moving this into pyobjectrc.rs, so we can access the inner value directly might be better.